docs/howtos/port-linkml.md

```diff
@@ -0,0 +1,293 @@
+from linkml_runtime.utils.metamodelcore import Bool, XSDDate
+...
+
+@dataclass
+class Event(YAMLRoot):
+    ...
+    started_at_time: Optional[Union[str, XSDDate]] = None
+
+```
+
+the python generator will also generate subclasses of builtin Python strings - these are useful for non-inlined references to type the reference slot: 
+
+```python
+from linkml_runtime.utils.yamlutils import YAMLRoot, extended_str, extended_float, extended_int
+
+class NamedThingId(extended_str):
+    pass
+
+class PersonId(NamedThingId):
+    pass
+
+...
+@dataclass
+class FamilialRelationship(Relationship):
+    ...
+    type: Union[str, "FamilialRelationshipType"] = None
+    related_to: Union[str, PersonId] = None
+
+```
+
+All of these decisions lead to a very fully featured set of generated data access classes, but these decisions do not need to be replicated when writing a generator for your target language.
+
+### Case study: typescriptgen
+
+In contrast to the python approaches, this targets typescript *Interfaces* as the construct of choice in the target language
+
+Note that interfaces are "compiled away" but are used for static analyis and IDE support
+
+### Case study: javagen
+
+Currently the approaches here are under discussion. There are different choices of target language construct:
+
+- classes
+- interfaces
+- records (introduced in java 16)
+
+The current pattern for generation roughly follows the pydanticgen approach of generating largely standalone classes (soon: records)
+
+### Implementing for your target language
+
+We recommend starting with a generator for simple standalone classes. These do not need to be as fully featured as pythongen. You can treat things like annotating
+model elements with URIs, providing introspection support, and generating JSON as separate concerns -- at least at first.
+
+## Step 2: Generate data access classes for the metamodel
+
+This will give you the equivalent of meta.py, types,py in linkml-runtime.
+
+E.g if your target langauge is C, and you decided on structs as target construct, then you will have structs with names like `ClassDefinition`, `SchemaDefinition`
+
+Depending on the approach you took in Step 1, this may also include an approach to instantiate this target language datamodel from YAML or JSON.
+
+If you have that then you should have the ability to do the equivalent of:
+
+```
+from linkml_runtime.loaders import yaml_loader
+from linkml_runtime.linkml_model.meta import SchemaDefinition
+
+schema = yaml_loader.load('my-schema.yaml', target=SchemaDefinition)
+# now do things with the schema
+```
+
+## Step 3: Implement core logic for derived schemas
+
+By this stage you can load and dump schema objects in a type-safe way. It may be tempting to just go off and implement things with this.
+
+However, you will also want to implement the core logic of LinkML as defined in the LinkML specification, under "Derived Schemas". You can do this simply by implementing the spec in your target language,
```

Comment:
Link to page for the "Derived Schemas"

Reply:
coming soon!  I am going to push this PR in, and we can make a note to add this when it is available at a stable URL.
for now it is here: https://github.com/linkml/linkml-model/blob/3e2101b3b17b4210427dc4eb46e105352d3d048a/specification/04derived-schemas.md